src/auth/hooks/useAuthProvider.ts

```diff
@@ -1,58 +1,144 @@
 import { IMessageContext } from "@saleor/components/messages";
-import { User } from "@saleor/fragments/types/User";
-import useLocalStorage from "@saleor/hooks/useLocalStorage";
+import { APP_DEFAULT_URI, APP_MOUNT_URI, DEMO_MODE } from "@saleor/config";
+import { useAuth, useAuthState } from "@saleor/sdk";
+import {
+  isSupported as isCredentialsManagementAPISupported,
+  login as loginWithCredentialsManagementAPI,
+  saveCredentials
+} from "@saleor/utils/credentialsManagement";
 import ApolloClient from "apollo-client";
-import { MutableRefObject } from "react";
+import { useEffect, useRef } from "react";
+import { useQuery } from "react-apollo";
 import { IntlShape } from "react-intl";
+import urlJoin from "url-join";
 
-import { useExternalAuthProvider } from "./useExternalAuthProvider";
-import { useSaleorAuthProvider } from "./useSaleorAuthProvider";
+import { userDetailsQuery } from "../queries";
+import {
+  ExternalLoginInput,
+  RequestExternalLoginInput,
+  RequestExternalLogoutInput,
+  UserContext
+} from "../types";
+import { UserDetails } from "../types/UserDetails";
+import { displayDemoMessage } from "../utils";
 
-export interface UseAuthProvider {
-  logout: () => void;
-  tokenAuthLoading: boolean;
-  tokenRefresh: () => Promise<boolean>;
-  tokenVerifyLoading: boolean;
-  user?: User;
-  autologinPromise?: MutableRefObject<Promise<any>>;
-}
 export interface UseAuthProviderOpts {
   intl: IntlShape;
   notify: IMessageContext;
   apolloClient: ApolloClient<any>;
 }
 
-export function useAuthProvider(opts: UseAuthProviderOpts) {
-  const [authPlugin, setAuthPlugin] = useLocalStorage("authPlugin", undefined);
+export function useAuthProvider({
+  intl,
+  notify,
+  apolloClient
+}: UseAuthProviderOpts): UserContext {
+  const {
+    login,
+    getExternalAuthUrl,
+    getExternalAccessToken,
+    logout,
+    setPassword
+  } = useAuth();
+  const { authenticated, authenticating } = useAuthState();
 
-  const saleorAuth = useSaleorAuthProvider({
-    authPlugin,
-    setAuthPlugin,
-    ...opts
-  });
+  const autologinPromise = useRef<Promise<any>>();
```

Comment:
Why are we passing a promise to context? If it's inside context it's only consumed by other react components, so why not use a state to check for resolve status?